DepotDownloader/ContentDownloader.cs

```diff
@@ -1295,5 +1281,49 @@ private static async Task DownloadSteam3Async(uint appId, List<DepotDownloadInfo
             }
 
         }
+
+        static void DumpManifestToTextFile( DepotDownloadInfo depot, ProtoManifest manifest )
+        {
+            StringBuilder manifestBuilder = new StringBuilder();
+            string txtManifest = Path.Combine( depot.installDir, string.Format( "manifest_{0}_{1}.txt", depot.id, depot.manifestId ) );
+            manifestBuilder.Append( string.Format( "Content Manifest for Depot {0}\n\n", depot.id ) );
+            manifestBuilder.Append( string.Format( "Manifest ID / date     : {0} / {1}\n", depot.manifestId, manifest.CreationTime ) );
+            
+            int numFiles = 0, numChunks = 0;
+            ulong uncompressedSize = 0, compressedSize = 0;
+
+            foreach ( var file in manifest.Files )
+            {
+                if ( file.Flags.HasFlag( EDepotFileFlag.Directory ) )
+                    continue;
+
+                numFiles++;
+                numChunks += file.Chunks.Count;
+
+                foreach ( var chunk in file.Chunks )
+                {
+                    uncompressedSize += chunk.UncompressedLength;
+                    compressedSize += chunk.CompressedLength;
+                }
+            }
+
+            manifestBuilder.Append( string.Format( "Total number of files  : {0}\n", numFiles ) );
+            manifestBuilder.Append( string.Format( "Total number of chunks : {0}\n", numChunks ) );
+            manifestBuilder.Append( string.Format( "Total bytes on disk    : {0}\n", uncompressedSize ) );
+            manifestBuilder.Append( string.Format( "Total bytes compressed : {0}\n", compressedSize ) );
+            manifestBuilder.Append( "\n          Size Chunks File SHA                                 Flags Name\n" );
+
+            foreach ( var file in manifest.Files )
+            {
+                manifestBuilder.Append( string.Format( "{0} {1} {2} {3} {4}\n",

```

Comment:
You can use string interpolation here, and make use of passing pad/format in it without the manual ToString/PadLeft calls.

I haven't tested it, but it should be something along the lines of this:
`$"{file.TotalSize:-14} {file.Flags:X-5}"`

Reply:
@xPaw I'm not sure it'll look all that nicer since you still need BitConverter for SHA-1 hash and you still need to cast flags to int from enum.

Reply:
`{file.TotalSize,14}"` or `"{0,14}", file.TotalSize`
`{file.Flags,5:X}"` or `"{0,5:X}", file.Flags` respectively.